pkg/service/service.go

```diff
@@ -1120,59 +1029,79 @@ func UpdateServicesWithOwnerReferences(client *kclient.Client, k8sComponents []d
 		if found {
 			continue
 		}
-		u.SetOwnerReferences(append(u.GetOwnerReferences(), ownerReference))
+		d.SetOwnerReferences(append(d.GetOwnerReferences(), ownerReference))
 
-		err = client.UpdateDynamicResource(group, version, resource, crdName, u)
+		err = client.UpdateDynamicResource(gvr.Resource.Group, gvr.Resource.Version, gvr.Resource.Resource, u.GetName(), d)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func getCRDName(crd map[string]interface{}) (string, bool) {
-	metadata, ok := crd["metadata"].(map[string]interface{})
-	if !ok {
-		return "", false
+func isLinkResource(kind string) bool {
+	return kind == "ServiceBinding"
+}
+
+// GetGVKFromUnstructured takes an unstructured struct and returns group, version and kind information as strings
+// It assumes group to be "core" for cases where metadata.apiVersion can't be split using forward slash, e.g. 'v1'
+func GetGVKFromUnstructured(u unstructured.Unstructured) schema.GroupVersionKind {
+	apiVersion := strings.Split(u.GetAPIVersion(), "/")
+	var group, version, kind string
+	if len(apiVersion) != 2 {
+		group = ""
+		version = apiVersion[0]
+	} else {
+		group = apiVersion[0]
+		version = apiVersion[1]
 	}
-	name, ok := metadata["name"].(string)
-	if !ok {
-		return "", false
+	kind = u.GetKind()
+	return schema.GroupVersionKind{
+		Group:   group,
+		Version: version,
+		Kind:    kind,
 	}
-	return name, true
 }
 
-func isLinkResource(kind string) bool {
-	return kind == "ServiceBinding"
-}
+// GetGVRFromUnstructured returns GVR from unstructured data
+func GetGVRFromUnstructured(client *kclient.Client, u unstructured.Unstructured) (*meta.RESTMapping, error) {
```

Comment:
This function has misleading name and comment. Name suggests that it should return GVR, but it returns RestMapping

Reply:
https://github.com/openshift/odo/pull/4964/commits/8270bb639dbd38589082f73b590955c489cdddd7